components/bounty/page-actions/view.tsx

```diff
@@ -0,0 +1,290 @@
+import React, { useState } from "react";
+import { isMobile, isTablet } from "react-device-detect";
+
+import { useTranslation } from "next-i18next";
+
+import EditIcon from "assets/icons/transactions/edit";
+
+import ConnectGithub from "components/connect-github";
+import { ContextualSpan } from "components/contextual-span";
+import CreatePullRequestModal from "components/create-pull-request-modal";
+import If from "components/If";
+import Modal from "components/modal";
+import MultiActionButton, { Action } from "components/multi-action-button";
+import ProposalModal from "components/proposal/create-proposal-modal";
+import Translation from "components/translation";
+import UpdateBountyAmountModal from "components/update-bounty-amount-modal";
+
+import { PageActionsViewProps } from "./page-actions";
+import PageActionsButton from "./page-actions-button/view";
+
+export default function PageActionsView({
+  bounty,
+  currentUser,
+  handleEditIssue,
+  handlePullrequest,
+  handleStartWorking,
+  isWalletConnected,
+  isGithubConnected,
+  isCreatePr,
+  isCreateProposal,
+  isExecuting,
+  showPRModal,
+  handleShowPRModal,
+  ghVisibility,
+  handleClickKyc,
+  isUpdateAmountButton,
+  isStartWorkingButton,
+  isKycButton,
+  isForkRepositoryLink,
+  isEditButton,
+  updateBountyData
+}: PageActionsViewProps) {
+  const { t } = useTranslation([
+    "common",
+    "pull-request",
+    "bounty",
+    "proposal",
+  ]);
+
+  const [showPRProposal, setShowPRProposal] = useState(false);
+  const [showGHModal, setShowGHModal] = useState(false);
+  const [showUpdateAmount, setShowUpdateAmount] = useState(false);
+
+  function handleActionPr(arg: {
+    title: string;
+    description: string;
+    branch: string;
+  }): Promise<void> {
+    if (!ghVisibility)
+      return new Promise((resolve) => resolve(setShowGHModal(true)));
+
+    return handlePullrequest(arg);
+  }
+
+  function handleActionWorking() {
+    if (!ghVisibility) return setShowGHModal(true);
+    handleStartWorking();
+  }
+
+  function ForkRepositoryLink() {
```

Comment:
all of these small functions _can_ be components, `page-action/actions/action-name.tsx`